pkg/datastore/datastore.go

```diff
@@ -31,39 +33,48 @@ type Writer interface {
 	// all package(s) from underlying datastore.
 	GetPackageIDs() string
 
-	// Write stores the list of operator manifest(s) into datastore
+	// Write stores the list of operator manifest(s) into datastore.s
 	Write(packages []*OperatorMetadata) error
 }
 
 // memoryDatastore is an in-memory implementation of operator manifest datastore.
 // TODO: In future, it will be replaced by an indexable persistent datastore.
 type memoryDatastore struct {
-	manifests   map[string]*OperatorMetadata
+	manifests   map[string]*OperatorManifest
 	unmarshaler blobUnmarshaler
 }
 
-func (ds *memoryDatastore) Read(packageID string) (*Manifest, error) {
-	metadata, exists := ds.manifests[packageID]
-	if !exists {
-		return nil, ErrManifestNotFound
-	}
+func (ds *memoryDatastore) Read(packageIDs []string) (*OperatorManifestData, error) {
+	data := StructuredOperatorManifestData{}
+	for _, packageID := range packageIDs {
+		manifest, exists := ds.manifests[packageID]
+		if !exists {
+			return nil, fmt.Errorf("package [%s] not found", packageID)
```

Comment:
Yes, but adding `best effort` here addresses a `faulty package` situation in part. Here is a suggestion of how we can take the following actions to make it holistic:
- When an `OperatorSource` is reconciled, if we come across a faulty package we ignore it. This will ensure that datastore has good packages only.
- We add a field called `Packages` in `OperatorSource.Status` to indicate the list of packages this operator source has successfully downloaded.  
- When a `CatalogSourceConfig` is reconciled, we should validate that the provided list of package(s) are in datastore, otherwise we throw an error and the object is moved to `Failed` phase. 

With these change(s), the system as a whole behaves consistently with faulty or non-existent package(s). We also need to implement the above at a `package` level, not at a `manifest` level.

Given this I believe it warrants a new PR.

Reply:
Sounds good. Please add a Jira story to track this.

Reply:
Jira story added.